core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/OrderByClause.kt

```diff
@@ -2,49 +2,47 @@ package ch.ergon.dope.resolvable.clause.model
 
 import ch.ergon.dope.DopeQuery
 import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.Resolvable
 import ch.ergon.dope.resolvable.clause.ISelectGroupByClause
 import ch.ergon.dope.resolvable.clause.ISelectOrderByClause
-import ch.ergon.dope.resolvable.expression.unaliased.type.Field
+import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.formatToQueryStringWithSymbol
-import ch.ergon.dope.validtype.StringType
 import ch.ergon.dope.validtype.ValidType
 
-enum class OrderByType(val type: String) {
+enum class OrderByType(val queryString: String) {
     ASC("ASC"),
     DESC("DESC"),
 }
 
 private const val ORDER_BY = "ORDER BY"
 
-open class SelectOrderByClause<T : ValidType>(private val stringField: Field<StringType>, private val parentClause: ISelectGroupByClause<T>) :
-    ISelectOrderByClause<T> {
-
+class OrderExpression(private val typeExpression: TypeExpression<out ValidType>, private val orderByType: OrderByType? = null) : Resolvable {
     override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
-        val parentDopeQuery = parentClause.toDopeQuery(manager)
-        val stringDopeQuery = stringField.toDopeQuery(manager)
+        val typeDopeQuery = typeExpression.toDopeQuery(manager)
         return DopeQuery(
-            queryString = formatToQueryStringWithSymbol(parentDopeQuery.queryString, ORDER_BY, stringDopeQuery.queryString),
-            parameters = parentDopeQuery.parameters.merge(stringDopeQuery.parameters),
+            queryString = "${typeDopeQuery.queryString}${orderByType?.queryString?.let { " $it" } ?: ""}",
+            parameters = typeDopeQuery.parameters,
         )
     }
 }
 
-class SelectOrderByTypeClause<T : ValidType>(
-    private val stringField: Field<StringType>,
-    private val orderByType: OrderByType,
+class SelectOrderByClause<T : ValidType>(
+    private val orderExpressions: List<OrderExpression>,
```

Comment:
Why do you pass a list instead of using vararg?

Reply:
i think i was talking about lists before i implemented this ticket, so i automaticly implemented it with lists as well.
i changed it to vararg.